lib/BackgroundJob/SyncSubmissionsWithLinkedFileJob.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * @copyright Copyright (c) 2024 Andrii Ilkiv <ailkiv@users.noreply.github.com>
+ *
+ * @author Andrii Ilkiv <ailkiv@users.noreply.github.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Forms\BackgroundJob;
+
+use OCA\Forms\Db\FormMapper;
+use OCA\Forms\Service\FormsService;
+use OCA\Forms\Service\SubmissionService;
+
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\BackgroundJob\IJobList;
+use OCP\BackgroundJob\QueuedJob;
+use OCP\Files\NotFoundException;
+use OCP\IUserManager;
+use OCP\IUserSession;
+
+use Psr\Log\LoggerInterface;
+use Throwable;
+
+class SyncSubmissionsWithLinkedFileJob extends QueuedJob {
+	public const MAX_ATTEMPTS = 10;
+
+	public function __construct(
+		private FormMapper $formMapper,
+		private FormsService $formsService,
+		private SubmissionService $submissionService,
+		ITimeFactory $time,
```

Comment:
I think it is better to have the inherited first

```suggestion
		ITimeFactory $time,
		private FormMapper $formMapper,
		private FormsService $formsService,
		private SubmissionService $submissionService,
```

Reply:
fixed